lib/util/mime_utils.js

```diff
@@ -96,6 +96,27 @@ shaka.util.MimeUtils = class {
     return codecs.split(',');
   }
 
+  /**
+   * Get the codec from a codec string, ignoring differences in base sample types declared in the fourCC.
+   *
+   * @param {string} codecString
+   * @return {string}
+   */
+  static getNormalizedCodec(codecString) {
+    switch (shaka.util.MimeUtils.getCodecBase(codecString)) {
```

Comment:
Is it possible to do the same for audio? For example: mp4a.40.2, mp4a.40.5, mp4a.40.29 and mp4a.40.42

Reply:
Related to https://github.com/shaka-project/shaka-player/issues/1567

Reply:
Hmm, maybe I am misunderstanding the question but all the codec string you quoted share the same fourCC so there is nothing to normalize, getNormalizedCodec will just return mp4a for all these.

Reply:
@joeyparrish what do you think?

Reply:
For mp4a.*, the result of getCodecBase is always "mp4a".  Is there some further mapping needed?

Or are you saying that those mp4a variants are actually mutually incompatible, and so should be mapped to different values here?

Reply:
the question is whether the decoder is the same and is compatible. if it is not compatible, even if they are aac, it should be prohibited….

Reply:
Yeah, okay, I get it now.  According to https://chromium.googlesource.com/chromium/src/media/+/master/base/mime_util_internal.cc, mp4a could refer to any of AAC, MP3, XHE AAC, AC3, EAC3, DTS, or DTSXP2.  I'm not sure if AAC/XHEAAC or DTS/DTSXP2 or AC3/EAC3 are compatible (use the same decoder).

If we know what is needed here (I don't know for sure), we can add mappings for audio now.  If not, we can merge this for video and follow up later for mp4a audio.

Reply:
Maybe related to https://github.com/shaka-project/shaka-player/issues/1528

Reply:
Interesting, maybe we do need a codec enum like theirs (MimeUtil::*).
Let me know if you want to merge this as is, and I can improve this method in a follow up change.

Reply:
It will be important to figure out which mp4a variants use what decoder.

So far it seems that mp3 is obvious and independent, XHE-AAC is _probably_ the same decoder as AAC, and I'm less sure about whether AC3 decoder == EAC3 decoder or DTS decoder == DTSXP2 decoder.  I asked someone in Chrome Media and they weren't sure either.  Those enums may have been added to Chrome for Cast, so I'm reaching out to some folks who work on media for the Cast platform.

Once we have an answer, the change will be trivial to make.  So let's hold this PR briefly until we can fix both audio and video.

Reply:
Updated the PR with the changes we discussed, PTAL.